StacksAndQueues/LargestRectangleInHistogram.cpp

```diff
@@ -1 +1,26 @@
-
+int Solution::largestRectangleArea(vector<int> &hist) {
+    stack<int> s; 
+    int max_area = 0;
+    int tp;
+    int area_with_top; 
+    int i = 0; 
+    while (i < hist.size()) { 
+        if (s.empty() || hist[s.top()] <= hist[i]) 
+            s.push(i++);
+        else { 
+            tp = s.top();
+            s.pop();
+            area_with_top = hist[tp] * (s.empty() ? i : i - s.top() - 1); 
+            if (max_area < area_with_top) 
+                max_area = area_with_top; 
+        } 
+    }
+    while (s.empty() == false) { 
+        tp = s.top(); 
```

Comment:
https://github.com/cruxrebels/InterviewBit/blob/d3eb60a916a4632d22c73f7d30373676dce5d4a8/StacksAndQueues/LargestRectangleInHistogram.cpp#L19-L23

Could these exact repeated logic be merged inside the above while loop itself?

---
[Non-blocker] You can create another PR if you find out.